filebeat/docs/inputs/input-filestream.asciidoc

```diff
@@ -152,6 +152,33 @@ include::../inputs/input-filestream-file-options.asciidoc[]
 
 include::../inputs/input-filestream-reader-options.asciidoc[]
 
+[float]
+=== Metrics
+
+This input exposes metrics under the <<http-endpoint, HTTP monitoring endpoint>>.
+These metrics are exposed under the `/inputs` path. They can be used to
+observe the activity of the input.
+
+[options="header"]
+|=======
+| Metric                    | Description
+| `files_read_total`        | Total number of files read.
+| `files_closed_total`      | Total number of files closed.
+| `files_created_total`     | Total number of file created events detected.
+| `files_updated_total`     | Total number of file updated events detected.
+| `files_ignored_total`     | Total number of file events ignored.
+| `files_truncated_total`   | Total number of file truncated events detected.
+| `files_deleted_total`     | Total number of file deleted events detected.
+| `files_renamed_total`     | Total number of file renamed events detected.
```

Comment:
I did some testing and got an interesting behaviour, it's not related to you metrics implementation, it's a behaviour of filestream:
- when a file is renamed and the new name is part of the files that input is reading, then a renamed event is triggered
- when a file is renamed and the new name is not part of the files input is reading, then a delete event is triggered.

Probably it is worth documenting it somehow, otherwise those metrics can be confusing, especially if they end up being  used on default dashboards on Kibana.

Reply:
Good points. Do we even want to expose these metrics, then? If the behaviors don't align with the names of the events, then it's just going to cause confusion and potential SDHs, even if we document it. Do we know all of these "quirks", and is it safe to assume the behaviors differ between the copytruncate_prospector and regular prospector? I'll be honest, I'm not very familiar with this input, and it's a complicated one.

Reply:
> Do we even want to expose these metrics, then?

That's a good question. I totally agree in their current state they will only cause confusion and SDHs.

> Do we know all of these "quirks", and is it safe to assume the behaviors differ between the copytruncate_prospector and regular prospector? 

I believe we don't :/. The behaviour I described I discovered while testing your PR.

I'm very familiar with filestream, but not with this specific part of the code. I also don't know much about the internals of copytruncate_prospector and prospector. @rdner do you know more about those prospectors?

See my other comment: https://github.com/elastic/beats/pull/35529#issuecomment-1577117060, I believe we should have a wider discussion about those metrics.

Reply:
Based on what we've discussed, I'll remove the prospector-related metrics for now, since it will take more thought and planning to implement those properly. Regarding the other metrics, I agree we should have a discussion about if there's overlap with #33771 and how to move forward.

I still need to look into testing, and not sure how involved that would be. I'd like to keep it an integration test if possible, and do something similar to what was done for the httpjson input. I *think* we should be able to access the metrics programmatically, which should make it easier to test.

Reply:
The `httpjson` input tests it in a nice way! I was trying to read the metrics from the global registry, but instantiating the registry during testing makes more sense.

I believe with that you can easily test your implementation.